preview/MsixCore/msixmgr/AutoPlay.cpp

```diff
@@ -0,0 +1,577 @@
+    RETURN_IF_FAILED(explorerKey.CreateSubKey(eventHandlerRootRegKeyName.c_str(), KEY_WRITE, &handleEventRootKey));
+
+    RegistryKey handleEventKey;
+    RETURN_IF_FAILED(handleEventRootKey.CreateSubKey(autoPlayObject.handleEvent.c_str(), KEY_WRITE, &handleEventKey));
+
+    RETURN_IF_FAILED(handleEventKey.SetStringValue(autoPlayObject.generatedhandlerName.c_str(), L""));
+
+    RETURN_IF_FAILED(handlerKey.SetUInt32Value(L"DesktopAppX", 1));
+
+    if (autoPlayObject.autoPlayType == DesktopAppxContent)
+    {
+        RETURN_IF_FAILED(handlerKey.SetStringValue(L"InvokeProgID", autoPlayObject.generatedProgId));
+
+        RETURN_IF_FAILED(handlerKey.SetStringValue(L"InvokeVerb", autoPlayObject.id));
+
+        RegistryKey verbRootKey;
+        RETURN_IF_FAILED(BuildVerbKey(autoPlayObject.generatedProgId, autoPlayObject.id, verbRootKey));
+
+        if (autoPlayObject.dropTargetHandler.size() > 0)
+        {
+            RegistryKey dropTargetKey;
+            RETURN_IF_FAILED(verbRootKey.CreateSubKey(dropTargetRegKeyName.c_str(), KEY_WRITE, &dropTargetKey));
+
+            std::wstring regDropTargetHandlerBuilder;
+            regDropTargetHandlerBuilder.append(L"{");
+            regDropTargetHandlerBuilder.append(autoPlayObject.dropTargetHandler);
+            regDropTargetHandlerBuilder.append(L"}");
+
+            RETURN_IF_FAILED(dropTargetKey.SetStringValue(L"CLSID", regDropTargetHandlerBuilder.c_str()));
+        }
+        else
+        {
+            RETURN_IF_FAILED(verbRootKey.SetStringValue(L"AppUserModelID", autoPlayObject.appUserModelId.c_str()));
+
+            std::wstring resolvedExecutableFullPath = m_msixRequest->GetPackageDirectoryPath() + L"\\" + m_msixRequest->GetPackageInfo()->GetRelativeExecutableFilePath();
+            RETURN_IF_FAILED(verbRootKey.SetStringValue(L"PackageRelativeExecutable", resolvedExecutableFullPath.c_str()));
+
+            RETURN_IF_FAILED(verbRootKey.SetStringValue(L"Parameters", autoPlayObject.parameters.c_str()));
+
+            RETURN_IF_FAILED(verbRootKey.SetStringValue(L"ContractId", L"Windows.File"));
+
+            RETURN_IF_FAILED(verbRootKey.SetUInt32Value(L"DesiredInitialViewState", 0));
+
+            RETURN_IF_FAILED(verbRootKey.SetStringValue(L"PackageId", m_msixRequest->GetPackageFullName()));
+
+            RegistryKey commandKey;
+            RETURN_IF_FAILED(verbRootKey.CreateSubKey(commandKeyRegName.c_str(), KEY_WRITE, &commandKey));
+
+            RETURN_IF_FAILED(commandKey.SetStringValue(L"DelegateExecute", desktopAppXProtocolDelegateExecuteValue));
+        }
+    }
+    else if (autoPlayObject.autoPlayType == DesktopAppxDevice)
+    {
+        std::wstring regHWEventHandlerBuilder;
+        regHWEventHandlerBuilder.append(L"{");
+        regHWEventHandlerBuilder.append(autoPlayObject.hwEventHandler);
+        regHWEventHandlerBuilder.append(L"}");
+
+        RETURN_IF_FAILED(handlerKey.SetStringValue(L"CLSID", regHWEventHandlerBuilder.c_str()));

```

Comment:
>.c_str() [](start = 85, length = 8)

do you need this? SetStringValue should take in std::wstring as parameter,

Reply:
Yes, it should take std::wstring as the value parameter. Updated to remove c_str() at all places.